kore/src/Kore/Simplify/Condition.hs

```diff
@@ -168,15 +177,33 @@ simplifyPredicates sideCondition original = do
     let predicates =
             SideCondition.simplifyConjunctionByAssumption original
                 & fst . extract
-    simplifiedPredicates <-
+    simplifiedPredicates <- do
+        let eliminatedExists =
+                map (simplifyPredicateExistElim $ freeVariableNames original <> freeVariableNames sideCondition) $
+                    toList predicates
```

Comment:
This line is pretty long, it should be reformatted a bit.

I'm not entirely sure if it's necessary to avoid variable names in the sidecondition, but let's leave it like this for now. Could you add a TODO comment there about this?